### PR TITLE
Agency Modal: New design

### LINKED ISSUE #1601: Refactor border-radius rules in styles.css
We have a few different ways / places of specifying `border-radius`:

* Root variable [`--border-radius: calc(3rem / 16)`](https://github.com/cal-itp/benefits/blob/fix/1578-rounded-corners/benefits/static/css/styles.css#L38) (=3px), used for [`.signout-link`](https://github.com/cal-itp/benefits/blob/fix/1578-rounded-corners/benefits/static/css/styles.css#L432) and [`.form-container .form-control`](https://github.com/cal-itp/benefits/blob/fix/1578-rounded-corners/benefits/static/css/styles.css#L499)
* Direct [8px rule on `.card`](https://github.com/cal-itp/benefits/blob/fix/1578-rounded-corners/benefits/static/css/styles.css#L640)
* Root variable [`--modal-border-radius: 8px;`](https://github.com/cal-itp/benefits/blob/fix/1578-rounded-corners/benefits/static/css/styles.css#L666), overridden for wider screens to [`--modal-border-radius: 4px;`](https://github.com/cal-itp/benefits/blob/fix/1578-rounded-corners/benefits/static/css/styles.css#L672), used for [`.modal-content`](https://github.com/cal-itp/benefits/blob/fix/1578-rounded-corners/benefits/static/css/styles.css#L690)
* The newest one from #1594

_Originally posted by @thekaveman in https://github.com/cal-itp/benefits/pull/1594#discussion_r1280887956_

### LINKED ISSUE #1634: Agency Selector modal design changes
Make changes to agency selector modal code as necessary to fit new design as spec'd out here #1444 
Ticket should be done after #1631 #1588 

## Needs
- Need to get the correctly sized logos from Figma (or Design)

## Acceptance criteria
- All logos are hard-coded to the new Desktop and Mobile dimensions
- SBMTD logo should fit in the new space
- Ensure spacing around the titles, X, etc. are also all correct
- If it makes sense, combine `modal` template with `modal-info` template so we can just have 1 template. (But if the spacing is _really_ off and different between the two... as it stands now in Figma, then perhaps, it makes sense to have 2 templates).

### Mobile

Image

Image

### Desktop
Image

Image

Image

Image

### PR DESCRIPTION
**Pull request by machikoyasuda**
closes #1634 
closes #1601 - rem-ify all of the border-radiuses

## What this PR does

image

- [x] Implement new Agency Modal design. 3 fits in 1 row.
- [x] Refactors `modal` and `modal-info` templates into 1 template. 
- [x] Modal close button SVG
- [x] Modal close button focus ring padding
- [x] Modal close button hover color

**Note:** When we get the 4th agency, we'll have to edit some CSS so that only 2 agencies show up in each row. That'll just have to happen when we get there.

### Desktop
**- Image dimension changes:** `148x72`
**- Card dimension changes:** Constant static width and height of `200x178`. No longer dynamic.
image

**- Card inner spacing changes:** Padding is now `16px`. Padding on Figma is wider, but 16px is the largest padding possible for SacRT name to fit.
image

**- Card outer spacing changes:** Gap between buttons is `44px`
image

### Mobile
- Image dimension changes: `52x36`
- Card dimension changes: Height of `73`, Width is 100% of the modal width minus 16px padding on left and right.
- Card inner spacing changes: Padding is `12px`.
- Card outer spacing changes: Gap between buttons is `16px`
- Space between the bottom of title and beginning of card row: `24px`

image

## Button Close X

- `#323A45` on load
- `#045B86` on hover

image

image

### CODE REVIEW

File: benefits/static/img/btn-close.svg
Comment (thekaveman) [inline benefits/static/img/btn-close.svg:1]: I don't think this image file is being used anywhere? It looks like a `data:` URL is used for the modal close buttons.

Comment (machikoyasuda) [reply benefits/static/img/btn-close.svg:1]: oh yea it's not!!

Comment (machikoyasuda) [reply benefits/static/img/btn-close.svg:1]: Used this site to do it: https://www.svgbackgrounds.com/tools/svg-to-css/

**Review by thekaveman — COMMENTED** [review]
This definitely looks correct for 2 and 3 agencies.

And while this isn't something we'll have to worry about probably for a bit, the modal does not look correct for 4 agencies:

**Figma**

image

**This PR**

image


Since we're doing this update, what do you think about future-proofing to the design we have now?

**Comment by machikoyasuda** [comment]
@thekaveman I wrote this up in the PR note up top:

`Note: When we get the 4th agency, we'll have to edit some CSS so that only 2 agencies show up in each row. That'll just have to happen when we get there.`

There is no way to write CSS Grid that can allow for logic like, "When there's 3 agencies, show them in 1 row. When there's 4 agencies, go with 2 rows of 2."

Once we get to 4 agencies, we have to just delete some CSS Flexbox Bootstrap classes, and then add this CSS Grid code  the `.card-row`:

```css
display: grid;
column-gap: 44px;
row-gap: 24px;
grid-template-columns: var(--card-width) var(--card-width);
justify-content: center;
```

**Comment by thekaveman** [comment]
@machikoyasuda 

> I wrote this up in the PR note up top

Sorry! I didn't see that.

**Review by thekaveman — APPROVED** [review]

**Pull request merged**